src/mpi/coll/src/csel.c

```diff
@@ -0,0 +1,1253 @@
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_LT:
+            nprintf("comm_size < %d\n", node->u.comm_size_lt.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_POW2:
+            nprintf("comm_size is power-of-two\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_ANY:
+            nprintf("comm_size is anything\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COMM_SIZE_NODE_COMM_SIZE:
+            nprintf("comm_size is the same as node_comm_size\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__AVG_MSG_SIZE_LE:
+            nprintf("avg_msg_size <= %d\n", node->u.avg_msg_size_le.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__AVG_MSG_SIZE_LT:
+            nprintf("avg_msg_size < %d\n", node->u.avg_msg_size_lt.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__AVG_MSG_SIZE_ANY:
+            nprintf("avg_msg_size is anything\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__TOTAL_MSG_SIZE_LE:
+            nprintf("total_msg_size <= %d\n", node->u.total_msg_size_le.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__TOTAL_MSG_SIZE_LT:
+            nprintf("total_msg_size < %d\n", node->u.total_msg_size_lt.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__TOTAL_MSG_SIZE_ANY:
+            nprintf("total_msg_size is anything\n");
+            break;
+        case CSEL_NODE_TYPE__CONTAINER:
+            nprintf("container\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COUNT_LE:
+            nprintf("count <= %d\n", node->u.count_le.val);
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COUNT_LT_POW2:
+            nprintf("count < nearest power-of-two less than comm size\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__COUNT_ANY:
+            nprintf("count is anything\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_SBUF_INPLACE:
+            nprintf("source buffer is MPI_IN_PLACE\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_BLOCK_REGULAR:
+            nprintf("all blocks have the same count\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_PARENT_COMM:
+            nprintf("communicator is the parent comm\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_NODE_CONSECUTIVE:
+            nprintf("process ranks are consecutive on the node\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_COMMUTATIVE:
+            nprintf("operation is commutative\n");
+            break;
+        case CSEL_NODE_TYPE__OPERATOR__IS_OP_BUILT_IN:
+            nprintf("other operators\n");
+            break;
+        default:
+            nprintf("unknown operator\n");
+            MPIR_Assert(0);
+    }
+
+    if (node->type != CSEL_NODE_TYPE__CONTAINER) {
+        print_tree(node->success);
```

Comment:
A side-comment: 

It would be nice if we can print out like `if-elif-elif-else` instead of nested multiple levels. In fact, it would be nice if we have an `if-elif-elif-else` kind of "combo" operators in the json.

EDIT: I now see the json input is like an `if-elif-elif-else`.

Reply:
We don't need to show any branches.  The print is very similar to the json format that we use.